test/js/visual.js

```diff
@@ -49,6 +49,62 @@ describe("Visual", function() {
             assert.strictEqual(imageElement.src.includes("frame=side-10"), true);
         });
 
+        it("should generate the correct image URL", () => {
+            const dom = new jsdom.JSDOM("<img id='image' />");
+            const imageElement = dom.window.document.getElementById("image");
+            const instance = new base.ripe.Ripe("myswear", "vyner");
+            const image = instance.bindImage(imageElement, {
+                brand: "myswear",
+                model: "vyner",
+                rotation: 20,
+                flip: true,
+                mirror: true,
+                boundingBox: [2000, 2000],
+                algorithm: "mask_top",
+                background: "ff00ff",
+                engine: "base",
+                initialsX: 2,
+                initialsY: 2,
+                initialsWidth: 20,
+                initialsHeight: 20,
+                initialsViewport: [1, 1, 20, 20],
+                initialsColor: "ff0000",
+                initialsOpacity: 0.7,
+                initialsAlign: "left",
+                initialsVertical: "top",
+                initialsRotation: 20,
+                initialsZindex: 5,
+                initialsAlgorithm: "multiplicative",
+                initialsBlendColor: "222222",
+                initialsImageRotation: 30,
+                initialsImageFlip: true,
+                initialsImageMirror: false,
+                debug: true,
+                fontWeight: 900,
+                fontSize: 50,
+                fontSpacing: 10,
+                fontTrim: true,
+                lineHeight: 30,
+                shadow: true,
+                shadowColor: "0000ff",
+                shadowOffset: 20,
+                offsets: { 0: [0, 10], 1: [-1, -10] },
+                curve: [
+                    [0.2, 0.2],
+                    [0.7, 0.2],
+                    [0.2, 0.5],
+                    [0.7, 0.5]
+                ]
+            });
+            assert.strictEqual(image._url, null);
+
+            image.update();
+            assert.strictEqual(
+                image._url,
+                "https://sandbox.platforme.com/api/compose?algorithm=mask_top&background=ff00ff&bounding_box=2000&bounding_box=2000&curve=%5B%5B0.2%2C0.2%5D%2C%5B0.7%2C0.2%5D%2C%5B0.2%2C0.5%5D%2C%5B0.7%2C0.5%5D%5D&debug=true&engine=base&flip=true&font_size=50&font_spacing=10&font_trim=true&font_weight=900&initials_algorithm=multiplicative&initials_align=left&initials_blend_color=222222&initials_color=ff0000&initials_height=20&initials_image_flip=true&initials_image_rotation=30&initials_opacity=0.7&initials_rotation=20&initials_vertical=top&initials_viewport=1&initials_viewport=1&initials_viewport=20&initials_viewport=20&initials_width=20&initials_x=2&initials_y=2&initials_z_index=5&line_height=30&mirror=true&offsets=%7B%220%22%3A%5B0%2C10%5D%2C%221%22%3A%5B-1%2C-10%5D%7D&rotation=20&shadow=true&shadow_color=0000ff&shadow_offset=20"

```

Comment:
Shouldn't the model and brand be here? This URL doesn't work:

![image](https://user-images.githubusercontent.com/3048875/99102486-07dbb380-25d6-11eb-80fb-c2a62b1282d6.png)

Reply:
It needs an brand and model and it works.
I think its intended for `_url` to not have as the test is made, so I think the test is correct.

If you check https://github.com/ripe-tech/ripe-sdk/pull/213 no changes related to brand and model were made.

Reply:
How can the test be correct if the URL you're checking produces a server error? I'm not saying you introduced any regression, but clearly there is something fishy here that needs fixing otherwise we have a test which isn't testing what it is supposed to.

Reply:
I will tweak it then